ERCS/erc-7913.md

```diff
@@ -0,0 +1,107 @@
+
+All these signature mechanism have one thing in common: they do not have a canonical ethereum address to represent them onchain. While users could deploy ERC-1271 compatible contract for each key individually, this would be cumbersome and expensive. As account abstraction tries to separate account addresses (that hold assets) from the key that control them, giving fixed onchain addresses to keys (and possibly sending assets to these address by mistake) is not the right approach. Instead, using a small number of verifier contracts that can process signature in a standard way, and having the accounts rely on these verifiers, feels like the correct approach. This has the advantage that once the verifier is deployed, any key can be represented using a `(verifier, key)` pair without requiring any setup cost.
+
+The `(verifier, key)` pairs can be given permission to control a smart account, to perform social recovery, or to do any other operation without ever having a dedicated onchain address. Systems that want to adopt this approach need to transition away from the model were signers are identified by their address to a new model where signers may not have an address, and are identified by `bytes` object.
+
+This definition is backward compatible with EOA and ERC-1271 contract: in that case the we use the address of the identity (EOA or contract) as the verifier and the key is empty.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Nomenclature
+
+- Keys are described as a `bytes` object of arbitrary length. For example a P256 or RSA public key.
+- Verifiers are smart contracts in charge of signature verification for a given key. They have an ethereum address.
+- A signer is a `bytes` object that is the concatenation of the verifier's address, and the key: `verifier || key`. A signer is at least 20 bytes long.
+
+### Key Verifier interface
+
+Verifiers MUST implement the following interface
+
+```solidity
+interface IERCXXXXKeyVerifier {
+  /**
+   * @dev Main function that verifies if `signature` is a valid signature by `key` of `hash`.
+   *
+   * MUST return the bytes4 magic value IERCXXXXKeyVerifier.isValidKeySignature.selector if the signature is valid.
+   * SHOULD return 0xffffffff or revert if the signature is not valid.
+   */
+  function isValidKeySignature(bytes calldata key, bytes32 hash, bytes calldata signature) external view returns (bytes4);
```

Comment:
We're using the verb "verification" so I would go with `verifySignature` or even just `verify` since this is a single-purpose contract we don't need to be maximally unambiguous.